include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -1762,8 +1762,14 @@ FEPointEvaluation<n_components_, dim, spacedim, Number>::prepare_evaluate_fast(
       else
         {
           if (renumber.empty())
-            for (unsigned int i = 0; i < dofs_per_component; ++i)
-              ETT::read_value(solution_values[i], comp, solution_renumbered[i]);
+            {
+              for (unsigned int i = 0; i < dofs_per_component; ++i)
+                ETT::read_value(
+                  solution_values[i + (component_in_base_element + comp) *
+                                        dofs_per_component],
```

Comment:
I think the compiler will not know that we have no integer overflow, so it will evaluate at least the addition in every loop iteration. I would prefer if we either create a `ScalarNumber *` variable to `solution_values.data() + (component_in_base_element + comp) * dofs_per_component` outside the loop or make sure the compiler uses `std::size_t` as the argument, i.e., a variable `std::size_t offset = (component_in_base_element + comp) * dofs_per_component` again defined outside the loop, as `std::size_t` signals the compiler that it cannot overflow because it is the same range of integer type as the addresses behind `solution_values`, and thus it can simply increment the pointer.

Reply:
I didn't see your comment and naturally went for the second way to make the code more readable. I will change the type to `std::size_t` as suggested.